### PR TITLE
Add JSON output for brig X get and list commands

### LINKED ISSUE #993: Add brig a "json" output
// Extracted from today's Brigade meeting

Brigade should have an alternative `json` output format that emits more machine-readable output. Machine-readable outputs are useful for building any tool that integrates with Brigade via brig due to the programming language used by the tool(Brigade as a go pkg is clearly unusable from tools written in other lang).

The U/X could be like `kubectl get -o json` but if someone has a better idea it would be fine to adopt.

Affected commands are:

- `brig build get`
- `brig build list`
- `brig project get`
- `brig project list`

### PR DESCRIPTION
**Pull request by radu-matei**
closes #993

- [x] brig build get
- [x] brig build list
- [x] brig project get
- [x] brig project list

```
$ brig build list --output json

[
    {
        "id": "01dqxcgfdjeffj97rm2sqqck72",
        "project_id": "brigade-b9f6da279ec59af95d121d1e1e502ecf056504e267f2f7dfc747eb",
        "type": "exec",
        "provider": "brigade-cli",
        "clone_url": "",
        "revision": {
            "commit": "",
            "ref": "master"
        },
        "payload": "",
        "script": "Y29uc3QgeyBldmVudHMsIEpvYiB9ID0gcmVxdWlyZSgiQGJyaWdhZGVjb3JlL2JyaWdhZGllciIpOwoKZXZlbnRzLm9uKCJleGVjIiwgKCkgPT4gewogICAgdmFyIGpvYiA9IG5ldyBKb2IoImRvLW5vdGhpbmciLCAiYWxwaW5lIik7CiAgICBqb2IudGFza3MgPSBbCiAgICAgICAgImVjaG8gSGVsbG8iLAogICAgICAgICJlY2hvIFdvcmxkIgogICAgXTsKCiAgICBqb2IucnVuKCk7Cn0pOwoKZXZlbnRzLm9uKCJjdXN0b20tZXZlbnQiLCAoKSA9PiB7CiAgICB2YXIgam9iID0gbmV3IEpvYigiY3VzdG9tLWpvYi1oYW5kbGVyIiwgImFscGluZSIpOwogICAgam9iLnRhc2tzID0gW2BlY2hvICJJIGFtIGRvaW5nIHZlcnkgaW1wb3J0YW50IHdvcmsiYF07CgogICAgam9iLnJ1bigpOwp9KTs=",
        "worker": {
            "id": "brigade-worker-01dqxcgfdjeffj97rm2sqqck72",
            "build_id": "01dqxcgfdjeffj97rm2sqqck72",
            "project_id": "brigade-b9f6da279ec59af95d121d1e1e502ecf056504e267f2f7dfc747eb",
            "start_time": "2019-10-23T15:41:30-07:00",
            "end_time": "2019-10-23T15:41:43-07:00",
            "exit_code": 0,
            "status": "Succeeded"
        }
    },
    {
        "id": "01dqxa1jc4610zea545h86ey5d",
        "project_id": "brigade-b9f6da279ec59af95d121d1e1e502ecf056504e267f2f7dfc747eb",
        "type": "exec",
        "provider": "brigade-cli",
        "clone_url": "",
        "revision": {
            "commit": "",
            "ref": "master"
        },
        "payload": "",
        "script": "Y29uc3QgeyBldmVudHMsIEpvYiB9ID0gcmVxdWlyZSgiQGJyaWdhZGVjb3JlL2JyaWdhZGllciIpOwoKZXZlbnRzLm9uKCJleGVjIiwgKCkgPT4gewogICAgdmFyIGpvYiA9IG5ldyBKb2IoImRvLW5vdGhpbmciLCAiYWxwaW5lIik7CiAgICBqb2IudGFza3MgPSBbCiAgICAgICAgImVjaG8gSGVsbG8iLAogICAgICAgICJlY2hvIFdvcmxkIgogICAgXTsKCiAgICBqb2IucnVuKCk7Cn0pOwoKZXZlbnRzLm9uKCJjdXN0b20tZXZlbnQiLCAoKSA9PiB7CiAgICB2YXIgam9iID0gbmV3IEpvYigiY3VzdG9tLWpvYi1oYW5kbGVyIiwgImFscGluZSIpOwogICAgam9iLnRhc2tzID0gW2BlY2hvICJJIGFtIGRvaW5nIHZlcnkgaW1wb3J0YW50IHdvcmsiYF07CgogICAgam9iLnJ1bigpOwp9KTs=",
        "worker": {
            "id": "brigade-worker-01dqxa1jc4610zea545h86ey5d",
            "build_id": "01dqxa1jc4610zea545h86ey5d",
            "project_id": "brigade-b9f6da279ec59af95d121d1e1e502ecf056504e267f2f7dfc747eb",
            "start_time": "2019-10-23T14:58:25-07:00",
            "end_time": "2019-10-23T14:58:39-07:00",
            "exit_code": 0,
            "status": "Succeeded"
        }
    }
]
```

### CODE REVIEW

File: brig/cmd/brig/commands/build_get.go [hunk 1]
```diff
@@ -47,6 +49,18 @@ func getBuild(out io.Writer, bid string) error {
...
+		if err != nil {
+			return err
+		}
+		_, err = out.Write(b)
+		return err
+	}
+
+	// TODO
+	//
+	// why are we setting these to nil?
```
Comment (radu-matei) [inline brig/cmd/brig/commands/build_get.go]: Here we set the script and payload displayed on the console to be nil.
Any idea why we do that for `build get` and not other commands?

Comment (radu-matei) [reply brig/cmd/brig/commands/build_get.go]: Once we figure this out, I can either modify the code or remove / improve the comment.

Comment (vdice) [reply brig/cmd/brig/commands/build_get.go]: Yea, it looks like they are removed for default output but then if the global verbose flag is set, it adds them back in: https://github.com/brigadecore/brigade/blob/master/brig/cmd/brig/commands/build_get.go#L58-L60

Perhaps we can add/improve the comment here?

File: brig/cmd/brig/commands/build_list.go [hunk 2]
```diff
@@ -79,27 +92,31 @@ func getBuilds(project string, client kubernetes.Interface, count int) ([]*build
...
 	})
 
+	if count == 0 || count > len(builds) {
+		count = len(builds)
+	}
+
+	return builds[:count], err
+}
+
+func getBuildsForStdoud(builds []*brigade.Build) []*buildForStdout {
```
[the remark this answers is not in the document]
Comment (radu-matei) [reply brig/cmd/brig/commands/build_list.go]: Good catch, thanks!

File: brig/cmd/brig/commands/build_get.go (hunk 1 above)
Comment (radu-matei) [reply brig/cmd/brig/commands/build_get.go]: Oh, the it was just below..
Ok, I added a comment there.

**Pull request merged**